crates/hir_expand/src/name.rs

```diff
@@ -4,6 +4,35 @@ use std::fmt;
 
 use syntax::{ast, SmolStr};
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct LifetimeName(SmolStr);
+
+impl fmt::Display for LifetimeName {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt::Display::fmt(&self.0, f)
+    }
+}
+
+impl LifetimeName {
+    pub const fn new_text(text: SmolStr) -> LifetimeName {
```

Comment:
This should be `pub(crate)` for the same reason `Name::new_text` is. 

I actually wonder if we need LifetimeName as a separate type at all. I guess we can just re-use `Name`? We lose a tiny bit of type safety, but I don't think lifetime and usual names can be confused?

Reply:
Right, reusing it doesn't hurt too much I guess. I was kinda stuck in the thought that lifetimes aren't ast::Names but that isn't be relevant here since it's not the ast at this point. 👍